bears/general/FilenameBear.py

```diff
@@ -74,12 +82,38 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        messages = []
```

Comment:
Up to you:
I have the feeling that this belongs before the part with
```
try:
    new_name = self._naming_conventions[...]
```
as there we start to analyze the filename.

But like said, up to you, as semantically the current code is totally correct.